app/controlplane/internal/service/attestation.go

```diff
@@ -268,11 +253,17 @@ func (s *AttestationService) storeAttestation(ctx context.Context, envelope *dss
 				func() error {
 					// reset context
 					ctx := context.Background()
-					digest, err := s.attestationUseCase.UploadToCAS(ctx, envelope, casBackend, workflowRunID)
+					var digest string
+					var err error
+					if bundle.GetDsseEnvelope() != nil {
+						digest, err = s.attestationUseCase.UploadBundleToCAS(ctx, bundle, casBackend, workflowRunID)
+					} else {
+						digest, err = s.attestationUseCase.UploadEnvelopeToCAS(ctx, envelope, casBackend, workflowRunID)
```

Comment:
since we are doing the fallback, we could technically stop storing the envelope in cas correct?

Reply:
At the end we have only one method that receives the []byte payload